### DISCUSSION TITLE
[BUG] Unable to Successfully Update Extensions

### DISCUSSION THREAD

**Discussion by mchippy** [SCCM-Zone/CM-SRS-Dashboards#32] [General]
Hi, 
I was trying to update the extensions for the dashboards via PowerShell and SSMS, but the query is unable to complete successfully. I'm not that advanced in SQL Server to determine the issue, but here are the returns I'm getting: 

### PowerShell: 
**RAN:** 
`.\Install-SRSReport.ps1 -ServerInstance '**********' -Database 'CM_***' -ExtensionsOnly -Overwrite -Verbose`

**RETURNED:** 
```
VERBOSE: Installation has started!
VERBOSE: Connecting to [CM_***]...
VERBOSE: Running SQL query...
VERBOSE: Connecting to [CM_***]...
VERBOSE: Running SQL query...
Error running query! Exception calling "Fill" with "1" argument(s): "Invalid column name 'RecurrenceType_MONTHLYBYWEEKDAYBASE'."
At C:\users\**********\Desktop\MEM.Zone-Dashboards_v5.0.4 (1)\MEM.Zone-Dashboards\Install-SRSReport.ps1:388 char:13
+             Throw (New-Object System.Exception("Error running query!  ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], Exception
    + FullyQualifiedErrorId : Error running query! Exception calling "Fill" with "1" argument(s): "Invalid column name 'RecurrenceType_MONTHLYBYWEEKDAYBASE'."
```

### SSMS
After uncommenting the lines and updating the placeholders, and running the script: 
```
Msg 207, Level 16, State 1, Procedure ufn_CM_GetNextMaintenanceWindow, Line 137 [Batch Start Line 41]
Invalid column name 'RecurrenceType_MONTHLYBYWEEKDAYBASE'.
```

**Comment by Ioan-Popovici** [comment]
I'll have a look, probably something simple to fix.
Thanks

**Comment by Ioan-Popovici** [comment]
Variable name bug, fixed in [v5.05](https://github.com/MEM-Zone/MEM.Zone-Dashboards/releases/tag/v5.0.5)

**Comment by mchippy** [comment]
I ran the updated SQL Query (v5.05) successfully, but two weeks later, the reports are still showing **No Service Window** at 100% of the environment. Other than the SQL query, is there anything else that needs to be updated? Like the report itself? 

![image](https://github.com/MEM-Zone/MEM.Zone-Dashboards/assets/97312277/acb4ca8a-0872-45f5-b4e9-c6cfef901a1e)

![image](https://github.com/MEM-Zone/MEM.Zone-Dashboards/assets/97312277/11b20b70-3425-478f-9668-549c6833bb01)

**Comment by Ioan-Popovici** [comment]
You will see the result instantly you don't need to wait.

I can see that you get the message 'SQL Helper Function `Missing`'
This could mean that you also need to assign rights to the stored procedure for the  ConfigMgr reporting group `smsschm_users`:

```sql
/* Grant select rights for this function to CM reporting users */
GRANT SELECT ON OBJECT::dbo.ufn_CM_GetNextMaintenanceWindow TO smsschm_users;
```

This is done by default when running the installer but must be done manually when adding the stored procedure through SSMS.

**Reply by mchippy** [reply]
That did the trick. Thanks!!

**Reply by mchippy** [reply]
What's interesting is the next maintenance window for a system that is actually scheduled for tomorrow is reporting as the maintenance window for next month. 

![image](https://github.com/MEM-Zone/MEM.Zone-Dashboards/assets/97312277/4d132bdf-8cca-4318-b834-d3a50dff2af3)

But the MW is set to four days after the second Tuesday of each month, which is tomorrow. Not a huge deal, and it may have something to do with the fact that we are so close to it.

**Reply by Ioan-Popovici** [reply]
No, it's a bug. Can you give me a screenshot from the MW config so I can replicate it in my environment?

**Reply by mchippy** [reply]
Sure! I have this applied to 18 servers in the environment, and these are the ones coming up as March 16th, which _is_ a maintenance window, but for the following month. 

![image](https://github.com/MEM-Zone/MEM.Zone-Dashboards/assets/97312277/34440706-b101-4160-a4bb-b1ef8edbbb0d)

**Reply by Ioan-Popovici** [reply]
Ok, I'll have a look to see what's happening.

Thanks 🙏

**Reply by mchippy** [reply]
Thank _YOU!_

**Comment by Ioan-Popovici** [comment]
If you're interested, I have a new version in the pipeline I can send you a preview version when it's ready.

**Reply by mchippy** [reply]
I need to get this into my lab because right now I'm using it for a client that had a request for much better reporting around software updates in ConfigMgr (which this offering definitely fits the bill!) but in my lab that sounds perfect. Let me know. :)
